### ISSUE TITLE
CMake no longer generates projects

### ISSUE THREAD

**Issue by aviumcaravan** [darlinghq/darling#1476]
**Expected Result**
CMake either finishing making a Makefile/build.ninja file or throwing an  error.

**Actual Result**
CMake throws deprecation warnings (if necessary) and then straight up freezes. no log is generated.

**Steps To Reproduce**
1. install CMake from source, pkgin or directly from CMake's dmg.
2. try to generate any project with any build system.

**System Information**
What system are you using?

| Software | Version |
| --- | --- |
| Distro | Arch Linux on AMD64 (updated 14th Jan 2024) |
| Linux Kernel | 6.6.10-arch1-1 |
| Darling | 5ee6a68f1 |

**Comment by CuriousTommy** [comment]
To make sure this isn't a CMake issue... If you build another CMake project, does it generate fine?

**Comment by aviumcaravan** [comment]
no, no projects are generated inside of Darling. this quite literally broke across a commit or two, same CMake binary.

**Comment by CuriousTommy** [comment]
> no, no projects are generated inside of Darling

What about non-darling projects that use CMake?

> this quite literally broke across a commit or two, same CMake binary.

So if you revert back to the older commit, CMake works fine? Can you identify the exact commit that works for you.

**Comment by aviumcaravan** [comment]
i think you misunderstood me. to clarify, what i mean is that CMake doesn't work *inside of the container*. the commit that worked is `c28e63693`

**Comment by CuriousTommy** [comment]
> i think you misunderstood me. to clarify, what i mean is that CMake doesn't work inside of the container.

Oh! I see what you mean now.

**Comment by CuriousTommy** [comment]
Hey @Fancy2209, I'm thinking of reverting your 882fec31025fc56dc5ce0a1310bbeebc1453847e, since I think these stubs need to be implemented. Just want to give you a heads up.

**Comment by CuriousTommy** [comment]
@aviumcaravan Can you test the `revert_SemaphoreStubs` branch.

**Comment by Fancy2209** [comment]
> Hey @Fancy2209, I'm thinking of reverting your [882fec3](https://github.com/darlinghq/darling/commit/882fec31025fc56dc5ce0a1310bbeebc1453847e) commit, since I think these stubs need to be implemented (and I won't have time to implement them). Just want to give you a heads up.

If it's needed I'm completely fine with that, thanks for letting me know

**Comment by aviumcaravan** [comment]
> @aviumcaravan Can you test the `revert_SemaphoreStubs` branch. Let me know if it fixes your issue.

yeah, gimme a shortie

**Comment by aviumcaravan** [comment]
it doesn't build. welp.
